### ISSUE TITLE
Native Security Integration

### ISSUE THREAD

**Issue by Iristyle** [hubotio/hubot#399]
From https://github.com/github/hubot/pull/384
## tldr;

Make role-based auth a core part of Hubot so commands can easily be restricted.

To add security to scripts, I currently make one-off mods of what's in `hubot-scripts`.  This means it's now more work for me to keep these in sync as they are bugfixed or have features added.
### Key points
- This should not be a breaking change to existing scripts
- `Help` should only list commands accessible to a given users role

I can see this working a couple of ways...
### Unique ids + management within Hubot :+1:

The best option I could come up with... 

If the commands had ids, theoretically additional Hubot commands could be used to tweak command security from within Hubot (define a few admin users through env variables).  The security configuration would then simply be stored as metadata in the brain.  Not sure how the ids would be generated though if existing scripts aren't modified (maybe file name + regex as a "key").  I don't think filename is enough because there are many commands in each script file.
### Global environment variable mapping :-1:

If the mapping of role to command could be established in a catch-all environment variable, then the scripts themselves wouldn't have to know anything about using roles.  This probably isn't a great idea though because this would become a monstrous / unwieldy configuration bit.

Like the above, this would require some sort of `primary key` on each command to identify it.  
### Opt-in  :-1:

For scripts that care, it could makes sense to hook up the roles at the same point as `robot.respond`, so that it's easy to just ignore a command if the user issuing it doesn't have the proper roles.  Scripts could hardcode the roles they want for particular commands, or provide environment variables.  This could be a real pain if every script decides that every command should have an env variable for it's roles.
### What I'm doing now

I took the [auth](https://github.com/github/hubot-scripts/blob/master/src/scripts/auth.coffee) script from hubot-scripts and made some tweaks to allow passing a list of roles instead of just a single role when making security assertions. I have a script that listens for `loaded` on the brain, then attaches roles to the users at start-up.  I've tweaked the scripts I need tweaked to add role based security.  Unfortunately this means that some of the scripts are now modified versions of what's in `hubot-scripts` --- which is the main problem I'd like to see solved here.
